### PR TITLE
Implement actual Jackson cached deserializer

### PR DESCRIPTION
**Pull request by pawelkowalski92**
With basic README 😉

### CODE REVIEW

File: json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/CachedDeserializationModule.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package pl.allegro.tech.jsoncache.jackson;
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.Module;
+
+public class CachedDeserializationModule extends Module {
+
+    private static final String MODULE_NAME = "cached-deserialization";
+    private static final Version VERSION = new Version(1, 0, 0, null, "pl.allegro.tech", "json-cache-jackson");
```
Comment (pawel-guzek) [inline json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/CachedDeserializationModule.java:9]: Is there a way to make this automatic (maybe retrieve the version from manifest at runtime?)? 
Many of contributors will forget to bump this

It is ok for now- but `Issue` to resolve this problem should be made

File: json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/keybuilder/strategy/JsonComponentExtractingStrategy.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import pl.allegro.tech.jsoncache.keybuilder.KeyPartMissingException;
+import pl.allegro.tech.jsoncache.keybuilder.stategy.KeyComponentBasedStrategy;
+
+import java.util.Optional;
+
+public class JsonComponentExtractingStrategy extends KeyComponentBasedStrategy<JsonNode> {
+
+    @Override
+    protected String extractKeyPart(JsonNode value, String component) throws KeyPartMissingException {
+        return Optional.ofNullable(value.get(component))
```
Comment (pawel-guzek) [inline json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/keybuilder/strategy/JsonComponentExtractingStrategy.java:13]: As far, as I've checked `value` shouldn't be null, so it is ok

**Review by pawel-guzek — APPROVED** [review]
Nicely done!
I left some food for thought :)

File: json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/CachedDeserializationModule.java (hunk 1 above)
Comment (pawelkowalski92) [reply json-cache-jackson/src/main/java/pl/allegro/tech/jsoncache/jackson/CachedDeserializationModule.java:9]: Fully agree ♥️ 
I guess we could enable some resource processor that prepares a proper build file and simply read it here.

**Pull request merged**